pkg/util/prefetch/reader.go

```diff
@@ -88,7 +100,12 @@ func (r *Reader) Read(data []byte) (int, error) {
 	}
 }
 
-// Close implements io.Closer.
+// Close implements io.Closer. Close should not be called concurrently with Read.
```

Comment:
seems most io.Reader is thread unsafe

Reply:
io.Reader interface does not limit the invocation, for example, the reader of io.Pipe is thread safe. So I want to notice the caller about safety.